oar/cli/cmd_image_signed_check.py

```diff
@@ -1,41 +1,45 @@
 import click
 import logging
-from oar.core.config_store import ConfigStore
-from oar.core.exceptions import JenkinsHelperException
-from oar.core.jenkins_helper import JenkinsHelper
+import re
+import requests
+from bs4 import BeautifulSoup
 from oar.core.const import *
-from oar.core.notification_mgr import NotificationManager
 from oar.core.worksheet_mgr import WorksheetManager
 
+
 logger = logging.getLogger(__name__)
 
 @click.command()
 @click.pass_context
-@click.option("-n","--build_number",type=int, help="provide build number to get job status")
-def image_signed_check(ctx,build_number):
+def image_signed_check(ctx):
     """
     Check payload image is well signed
     """
-    # get config store from context
     cs = ctx.obj["cs"] 
-    jh = JenkinsHelper(cs) 
-    if not build_number:
-        report = WorksheetManager(cs).get_test_report()
-        image_signed_check_result = report.get_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY)
-        if image_signed_check_result == TASK_STATUS_PASS:
-            logger.info("image signed check already pass, not need to trigger again")
-        elif image_signed_check_result == TASK_STATUS_INPROGRESS:   
-            logger.info("job[signature_check] already triggered and in progress, not need to trigger again")
+    report = WorksheetManager(cs).get_test_report()
+    image_signed_check_result = report.get_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY)
+    if image_signed_check_result == TASK_STATUS_PASS:
+        logger.info("image signed check already pass, not need to trigger again")
+    else:
+        report.update_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_INPROGRESS)
+        res = requests.get(cs.get_release_url() + "/releasestream/4-stable/release/" + cs.release)
+        if res.status_code != 200:
+            print(res.status_code, res.reason)
+        soup = BeautifulSoup(res.content, 'lxml')
+        # 1. get the digest
+        digest_sha = soup.find(string=re.compile("sha256:"))
+        if len(digest_sha) == 0:
+            logger.error("No image digest found!")
+        print(digest_sha)
+        reformatted_digest = digest_sha.replace(':', '=')
+        # 2. query the mirror location
+        mirror_url = cs.get_signature_url() + reformatted_digest + "/"
+        print(f'Comparing digest from url: {mirror_url}')
+        res = requests.get(mirror_url)
```

Comment:
@wewang58 need to catch exception raised from 3rd party library, then mark the task as failed

Reply:
@rioliu-rh  Updated, please check again, thanks